### PR TITLE
Issue #7667: Update doc for MissingCtor

### LINKED ISSUE #7667: Update doc for MissingCtor
https://checkstyle.org/config_coding.html#MissingCtor

Add example for each existing config to show what code is ok and what code cause violation.
Good example is https://checkstyle.org/config_naming.html#AbstractClassName
Look at referenced commits of #5832, to see what and how should be changed in code.

Please play with module/check by means of https://checkstyle.org/cmdline.html to understand how it works.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by Gaurav-Punjabi**
Fixes #7667 

#### 1. Screenshots
Screen Shot 2020-04-27 at 11 15 36 PM

#### 2. Output
```
gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="MissingCtor"/>
    </module>
</module>gauravpunjabi@Gauravs-MacBook-Pro:~/Coding/Open Source/checkstyle$  java -jar target/checkstyle-8.33-SNAPSHOT-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /Users/gauravpunjabi/Coding/Open Source/checkstyle/Test.java:8:1: Class should define a constructor. [MissingCtor]
Audit done.
Checkstyle ends with 1 errors.

```

### CODE REVIEW

**Comment by Gaurav-Punjabi** [comment]
@strkkk @esilkensen @pbludov Please review.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java [hunk 1]
```diff
@@ -35,6 +35,22 @@
  * <pre>
  * &lt;module name=&quot;MissingCtor&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class ExampleOk { // OK
+ *   private int a;
+ *   ExampleOk(int a) {
+ *     this.a = a;
+ *       // Some other initialization here.
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java:44]: please remove this comment

File: src/xdocs/config_coding.xml [hunk 2]
```diff
@@ -3324,6 +3324,22 @@ class TestHashCode {
         <source>
 &lt;module name=&quot;MissingCtor&quot;/&gt;
         </source>
+        <p>Example:</p>
+        <source>
+class ExampleOk { // OK
+    private int a;
```
Comment (strkkk) [inline src/xdocs/config_coding.xml:3330]: nested elements indentation is 4, but it should be consistent with javadoc and be 2

Comment (Gaurav-Punjabi) [reply src/xdocs/config_coding.xml]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java (hunk 1 above)
Comment (Gaurav-Punjabi) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java]: Done

File: src/xdocs/config_coding.xml (hunk 2 above)
Comment (Gaurav-Punjabi) [reply src/xdocs/config_coding.xml]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java [hunk 3]
```diff
@@ -35,6 +35,21 @@
...
  * &lt;module name=&quot;MissingCtor&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * class ExampleOk { // OK
+ *   private int a;
+ *   ExampleOk(int a) {
+ *     this.a = a;
+ *   }
+ * }
```
Comment (strkkk) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java:45]: please add one more ok example with no-arg constructor (overriding a default one)

Comment (Gaurav-Punjabi) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/MissingCtorCheck.java:45]: Done

**Comment by strkkk** [comment]
ok to merge after ci passes

**Pull request merged**